src/packbeam_api.erl

```diff
@@ -707,21 +707,43 @@ strip_padding(Data) ->
     Data.
 
 %% @private
-uncompress_literals(Chunks) ->
+maybe_uncompress_literals(Chunks) ->
     case proplists:get_value("LitT", Chunks) of
         undefined ->
             {Chunks, undefined};
-        <<_Header:4/binary, Data/binary>> ->
-            UncompressedData = zlib:uncompress(Data),
-            {
-                lists:keyreplace(
-                    "LitT",
-                    1,
-                    Chunks,
-                    {"LitU", UncompressedData}
-                ),
-                UncompressedData
-            }
+        LitT ->
+            <<_Header:4/binary, Data/binary>> = LitT,
+            case literals_need_uncompress(LitT) of
+                true ->
+                    do_uncompress_literals(Chunks, Data);
+                false ->
+                    {Chunks, Data}
+            end
+    end.
+
+%% @private
+do_uncompress_literals(Chunks, Data) ->
+    UncompressedData = zlib:uncompress(Data),
+    {
+        lists:keyreplace(
+            "LitT",
+            1,
+            Chunks,
+            {"LitU", UncompressedData}
+        ),
+        UncompressedData
+    }.
+
+%% @private
+literals_need_uncompress(LitT) ->
+    <<_Offset:8/binary, _Skip:7/binary, Size:1/binary, _Data/binary>> = LitT,
```

Comment:
I'm not sure this is exactly what @bettio suggested, bad things would happen if there are 256 literals.

Reply:
This was my mistake. Thanks for the help.